api/src/main/java/org/pmiops/workbench/api/WorkspacesController.java

```diff
@@ -279,10 +331,25 @@ private FirecloudWorkspaceId generateFirecloudWorkspaceId(String namespace, Stri
     Timestamp now = new Timestamp(clock.instant().toEpochMilli());
     dbWorkspace.setLastModifiedTime(now);
     // TODO: add version, check it here
-    dbWorkspace = workspaceService.dao.save(dbWorkspace);
+    dbWorkspace = workspaceService.getDao().save(dbWorkspace);
     return ResponseEntity.ok(TO_CLIENT_WORKSPACE.apply(dbWorkspace));
   }
 
+  @Override
+  public ResponseEntity<EmptyResponse> shareWorkspace(String workspaceNamespace, String workspaceId,
+      UserRoleList userRoleList) {
+    Set<WorkspaceUserRole> dbUserRoles = new HashSet<WorkspaceUserRole>();
+    for (UserRole user : userRoleList.getItems()) {
+      WorkspaceUserRole newUser = new WorkspaceUserRole();
+      newUser.setUser(userDao.findUserByEmail(user.getUser()));
```

Comment:
You need to check for this being null (and throw a BadRequestException if that occurs.)

If we wanted this to be slightly more efficient we could look for all the users in the ACL in a single query:

userDao.findByEmailIn(<list of emails>)